### DISCUSSION TITLE
escaping new lines (`/n`) for sarif format

### DISCUSSION THREAD

**Discussion by DmitriyLewen** [community/community#80935] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello!
`sarif` docs say (about text property) `The SARIF producer SHALL escape any characters that JSON [[RFC8259](https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/sarif-v2.1.0-errata01-os-complete.html#RFC8259)] requires to be escaped.` - https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/sarif-v2.1.0-errata01-os-complete.html#_Toc141790674
And, if i undertand correctly, we need to escape `/`, so we need to use `//n` for new lines.

But after using `//n` and uploading with `github/codeql-action/upload-sarif@v2` we don't get new lines in UI:
https://github.com/aquasecurity/trivy/discussions/5804

Can you explain which way is correct?

Thank you in advance!
Best Regards, Dmitriy!

**Comment by jketema** [comment]
Hi @DmitriyLewen,

A concrete repoducer where you're seeing this - and what exactly you're seeing after the fix on your side - would help us diagnose this issue.

**Reply by DmitriyLewen** [reply]
Hello @jketema 
Thanks for your answer!

I created test repo for this issue- https://github.com/DmitriyLewen/test-sarif-upload/security/code-scanning/1
I expect new lines instead of `/n` 

You can see sarif file here - https://github.com/DmitriyLewen/test-sarif-upload/blob/main/report.sarif.json

**Reply by jketema** [reply]
Thanks. I'll try to loop in someone from the right team.

**Reply by DmitriyLewen** [reply]
Thanks a lot!

**Reply by orhantoy** [reply]
Have you tried `\n` (instead of `\\n`)?

**Reply by DmitriyLewen** [reply]
`\n` works correctly, but i try to understand correct way.
docs say that we need to escape `\`.

**Reply by orhantoy** [reply, marked as answer]
The docs also specifically calls out line breaks though:

> Notwithstanding any necessary transcoding and escaping, the SARIF producer SHALL preserve the text artifact’s line breaking convention (for example, "\n" or "\r\n").

**Reply by DmitriyLewen** [reply]
So we don't need to escape `\r\n` and `\n`, right?

**Reply by orhantoy** [reply]
Correct.

**Reply by DmitriyLewen** [reply]
Got it!
Thanks for your answer!

**Reply by DmitriyLewen** [reply]
@orhantoy 1 more small question:
do we need to use escaping for `markdown` property?

**Reply by orhantoy** [reply]
No. I think the JSON escaping clarification in the docs mainly concerns quotes, so something like double quotes would need to be escaped.

**Reply by DmitriyLewen** [reply]
Thank you again!
